### PR TITLE
Add per type scores in seqeval metric

### PR DESCRIPTION
**Pull request by jplu**
This PR add a bit more detail in the seqeval metric. Now the usage and output are:

```python
import nlp
met = nlp.load_metric('metrics/seqeval')
references = [['O', 'O', 'O', 'B-MISC', 'I-MISC', 'I-MISC', 'O'], ['B-PER', 'I-PER', 'O']]
predictions =  [['O', 'O', 'B-MISC', 'I-MISC', 'I-MISC', 'I-MISC', 'O'], ['B-PER', 'I-PER', 'O']]
met.compute(predictions, references)

#Output: {'PER': {'precision': 1.0, 'recall': 1.0, 'f1': 1.0, 'number': 1}, 'MISC': {'precision': 0.0, 'recall': 0.0, 'f1': 0, 'number': 1}, 'overall_precision': 0.5, 'overall_recall': 0.5, 'overall_f1': 0.5, 'overall_accuracy': 0.8}
```

It is also possible to compute scores for non IOB notations, POS tagging for example hasn't this kind of notation. Add `suffix` parameter:

```python
import nlp
met = nlp.load_metric('metrics/seqeval')
references = [['O', 'O', 'O', 'MISC', 'MISC', 'MISC', 'O'], ['PER', 'PER', 'O']]
predictions =  [['O', 'O', 'MISC', 'MISC', 'MISC', 'MISC', 'O'], ['PER', 'PER', 'O']]
met.compute(predictions, references, metrics_kwargs={"suffix": True})

#Output: {'PER': {'precision': 1.0, 'recall': 1.0, 'f1': 1.0, 'number': 1}, 'MISC': {'precision': 0.0, 'recall': 0.0, 'f1': 0, 'number': 1}, 'overall_precision': 0.5, 'overall_recall': 0.5, 'overall_f1': 0.5, 'overall_accuracy': 0.9}
```

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
LGTM!

File: metrics/seqeval/seqeval.py [hunk 1]
```diff
@@ -38,19 +40,110 @@
...
+        type_: current type.
+    Returns:
+        chunk_end: boolean.
+    """
+    chunk_end = False
+
+    if prev_tag == 'E': chunk_end = True
+    if prev_tag == 'S': chunk_end = True
+
+    if prev_tag == 'B' and tag == 'B': chunk_end = True
```
Comment (patrickvonplaten) [inline metrics/seqeval/seqeval.py:76]: Could we maybe replace all those if lines with `if prev_tag in ["B", "I"] and tag in ["B", "S", "O"]` or prev_tag in ["E", "S"]? I think this covers all cases

File: metrics/seqeval/seqeval.py [hunk 2]
```diff
@@ -38,19 +40,110 @@
...
+        type_: current type.
+    Returns:
+        chunk_start: boolean.
+    """
+    chunk_start = False
+
+    if tag == 'B': chunk_start = True
+    if tag == 'S': chunk_start = True
+
+    if prev_tag == 'E' and tag == 'E': chunk_start = True
```
Comment (patrickvonplaten) [inline metrics/seqeval/seqeval.py:104]: same here this can be made more readable with list look-up no?

File: metrics/seqeval/seqeval.py (hunk 1 above)
Comment (jplu) [reply metrics/seqeval/seqeval.py:76]: Thanks! I'm looking at it!

Comment (jplu) [reply metrics/seqeval/seqeval.py]: Does-it look better now?

Comment (patrickvonplaten) [reply metrics/seqeval/seqeval.py]: great!

**Comment by thomwolf** [comment]
Can you put the kwargs as normal kwargs instead of a dict? (And add them to the kwargs description As well)

**Comment by jplu** [comment]
@thom Is-it what you meant?

**Comment by thomwolf** [comment]
Yes and there is a dynamically generated doc string in the metric script KWARGS DESCRIPTION

**Pull request merged**